src/tests/intg/sss_netgroup_thread_test.c

```diff
@@ -0,0 +1,60 @@
+#define _GNU_SOURCE /* for pthread_yield */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <pthread.h>
+#include <pwd.h>
+#include <grp.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <sys/types.h>
+
+
+
+struct data {
+    char *group;
+    char *host;
+    char *user;
+    char *domain;
+    bool *failed;
+};
+
+static void *full_netgroup(void *arg)
+{
+    int ret;
+    size_t c = 0;
+    struct data *data = arg;
+
+    do {
+        ret = innetgr(data->group, data->host, data->user, data->domain);
+        if (ret != 1) {
+            *(data->failed) = true;
+        }
+        c++;
+    } while (!*(data->failed) && c<100000);
```

Comment:
Formally, this (access to shared memory location `&failed`) isn't thread-safe.

I think the simplest solution would be to use `bool failed[2]`